### PR TITLE
Add SSE component

### PR DESCRIPTION
**Pull request by snehilvj**
This PR adds ServerSentEvents components which enables dash apps to subscribe to server sent events.

**server.py**
```python
import asyncio
import random

import uvicorn
from sse_starlette.sse import EventSourceResponse
from starlette.applications import Starlette
from starlette.middleware import Middleware
from starlette.middleware.cors import CORSMiddleware

middleware = [Middleware(CORSMiddleware, allow_origins=['*'])]
app = Starlette(debug=True, middleware=middleware)


async def numbers():
    while True:
        await asyncio.sleep(1)
        yield [random.randrange(200, 1000) for _ in range(10)]


@app.route("/")
async def sse(request):
    generator = numbers()
    return EventSourceResponse(generator)


if __name__ == "__main__":
    uvicorn.run(app, port=8000, log_level='info')
```

**client.py**
```python
from dash import Dash, html, Input, Output, dcc
from dash_extensions import ServerSentEvents

app = Dash(__name__)

app.layout = html.Div(
    [
        dcc.Graph(id="graph"),
        ServerSentEvents(id="sse", url="http://127.0.0.1:8000"),
    ]
)


app.clientside_callback(
    """
    function(val) {
        values = val ? JSON.parse(val) : []
        return {data: [{y: values , type: "scatter"}]}
    }
    """,
    Output("graph", "figure"),
    Input("sse", "message"),
)


if __name__ == "__main__":
    app.run_server(debug=True)
```

**Output:**

https://user-images.githubusercontent.com/91216500/152369583-8926e593-13d0-4e39-b639-f5f1daec3e20.mov

Please let me know if this PR needs more changes.

### CODE REVIEW

**Comment by emilhe** [comment]
It look really cool! Any reason, the component is not just called `EventSource`?

**Comment by snehilvj** [comment]
I named the component according to what it conceptually does. DashEventSource makes sense only if you know how SSE is implemented in JavaScript, which is via creating an event source. What do you think?

**Comment by emilhe** [comment]
I generally like to link the component names to the JS ancestors as close as possible where it makes sense. So I am learning more towards `DashEventSource`. Or maybe `ServerEventSource` (or `DashServerEventSource`, but that's a bit long...).

**Comment by emilhe** [comment]
I think we should also add a `readyState` property before releasing the component (but I could also do that).

File: src/lib/components/ServerSentEvents.react.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/** An interface to server sent events in Dash */
+export default class ServerSentEvents extends React.Component {
+    componentDidMount() {
+        const { url, withCredentials, setProps } = this.props;
+
+        this.eventSource = new EventSource(url, { withCredentials });
+
+        this.eventSource.onmessage = (event) => {
+            setProps({ message: event.data });
```
Comment (emilhe) [inline src/lib/components/ServerSentEvents.react.js:12]: Is there a reason why you return the event data for messages, but a JSON object for errors?

Comment (emilhe) [reply src/lib/components/ServerSentEvents.react.js:12]: (I guess it would be most coherent to do one or the other)

Comment (snehilvj) [reply src/lib/components/ServerSentEvents.react.js:12]: I was not sure which properties to use for error event. Then I looked at the WebSocket component and borrowed from there.

**Pull request merged**